src/com/android/volley/toolbox/HttpHeaderParser.java

```diff
@@ -90,12 +90,18 @@
             softExpire = now + (serverExpires - serverDate);
         }
 
+        if (softExpire == 0) {
+            softExpire = Integer.parseInt(headers.get(Cache.Entry.KEY_CACHED_SOFTTTL));
+        }
+
+        long ttl = Integer.parseInt(headers.get(Cache.Entry.KEY_CACHED_TTL));
+
         Cache.Entry entry = new Cache.Entry();
         entry.data = response.data;
         entry.isImage = response.isImage;
         entry.etag = serverEtag;
         entry.softTtl = softExpire;
-        entry.ttl = entry.softTtl;
```

Comment:
wacky that before, the softtl and the ttl were being set to the same value. I wonder if we did that for permacache for some reason. Does the current implementation of Google Volley do this?